packages/indy-vdr/tests/indy-vdr-anoncreds-registry.e2e.test.ts

```diff
@@ -33,43 +57,55 @@ const agent = new Agent({
       indySdk,
     }),
     dids: new DidsModule({
+      registrars: [new IndyVdrIndyDidRegistrar()],
       resolvers: [new IndyVdrSovDidResolver(), new IndyVdrIndyDidResolver()],
     }),
   },
 })
 
-const indyVdrPoolService = agent.dependencyManager.resolve(IndyVdrPoolService)
+const indyVdrPoolService = endorser.dependencyManager.resolve(IndyVdrPoolService)
 const pool = indyVdrPoolService.getPoolForNamespace('pool:localtest')
 
 describe('IndyVdrAnonCredsRegistry', () => {
+  let endorserDid: string
   beforeAll(async () => {
-    await agent.initialize()
+    await endorser.initialize()
+    const unqualifiedSubmitterDid = await importExistingIndyDidFromPrivateKey(
+      endorser,
+      TypedArrayEncoder.fromString('00000000000000000000000Endorser9')
+    )
+    endorserDid = `did:indy:pool:localtest:${unqualifiedSubmitterDid}`
 
-    await importExistingIndyDidFromPrivateKey(agent, TypedArrayEncoder.fromString(publicDidSeed))
+    await agent.initialize()
   })
 
   afterAll(async () => {
     for (const pool of indyVdrPoolService.pools) {
       pool.close()
     }
 
+    await endorser.shutdown()
+    await endorser.wallet.delete()
     await agent.shutdown()
     await agent.wallet.delete()
   })
 
   // One test as the credential definition depends on the schema
-  test('register and resolve a schema and credential definition', async () => {
+  test('register and resolve a schema and credential definition (internal)', async () => {
     const dynamicVersion = `1.${Math.random() * 100}`
 
-    const legacyIssuerId = 'TL1EaPFCZ8Si5aUrqScBDt'
-    const signingKey = Key.fromPublicKeyBase58('FMGcFuU3QwAQLywxvmEnSorQT3NwU9wgDMMTaDFtvswm', KeyType.Ed25519)
-    const didIndyIssuerId = 'did:indy:pool:localtest:TL1EaPFCZ8Si5aUrqScBDt'
+    const legacyIssuerId = 'DJKobikPAaYWAu9vfhEEo5'
+    const didIndyIssuerId = 'did:indy:pool:localtest:DJKobikPAaYWAu9vfhEEo5'
+    const signingKey = await verificationKeyForIndyDid(agent.context, didIndyIssuerId)
 
-    const legacySchemaId = `TL1EaPFCZ8Si5aUrqScBDt:2:test:${dynamicVersion}`
-    const didIndySchemaId = `did:indy:pool:localtest:TL1EaPFCZ8Si5aUrqScBDt/anoncreds/v0/SCHEMA/test/${dynamicVersion}`
+    const legacySchemaId = `DJKobikPAaYWAu9vfhEEo5:2:test:${dynamicVersion}`
+    const didIndySchemaId = `did:indy:pool:localtest:DJKobikPAaYWAu9vfhEEo5/anoncreds/v0/SCHEMA/test/${dynamicVersion}`
 
-    const schemaResult = await indyVdrAnonCredsRegistry.registerSchema(agent.context, {
-      options: {},
+    const schemaResult = await indyVdrAnonCredsRegistry.registerSchema(endorser.context, {
+      options: {
+        endorserMode: 'internal',
+        endorserDid,
```

Comment:
I think we should add a test case where endorserMode is `internal`, but where the endorserDid is different from the issuerId. Can't see if that's the case here?